Friendly language names on linux

erikjanwestendorp opened the pull request:
**Update LanguageMapDefinition**
Apparently the Display Name on Linux is different. It shows the language code. Therefore, the mapper has been modified to use the English name

**Update the SaveLanguage method in the LanguageController**
The language name is stored in the database. To be consistent between Windows and Linux it is good to store the English name here too

Belongs to: https://github.com/umbraco/Umbraco-CMS/issues/12091

mikecp reviewed (changes requested):
Hello @erikjanwestendorp,

Thanks for this PR!

Besides @nul800sebastiaan remark, I would also like to mention that things should remain consistent when setting the CultureName property, so for example on line 189 of the LanguageController, we have the following code:

`existingById.CultureName = cultureAfterChange.DisplayName`;

which should then also be updated to

`existingById.CultureName = cultureAfterChange.EnglishName;`

It would be great if you could if there are other places where we set the CultureName property and adapt it accordingly  👍

ronaldbarendse commented:
Maybe it's time to revisit PR https://github.com/umbraco/Umbraco-CMS/pull/8384 for v9/v10, so the language name can be edited to what the implementor actually wants to display... Or otherwise at least check the changes to see where the name is assigned.

mikecp commented:
Another possibility I had been thinking of while checking #12055 was to display the language/Culture names at the different places in the actual user's Culture, and not always in English. I did not have time to go much further than the idea, but I had the intention to make the changes in that direction.

erikjanwestendorp commented:
@nul800sebastiaan @ronaldbarendse @mikecp Sorry for the delay on this one.

I updated the PR. I made a migration so that the language name is no longer stored in the database. In addition, I have also consistently implemented the use of the English name in several places.

I don't have a good solution for display the language name localized. Maybe @ronaldbarendse suggestion could be a solution or we have to work resources files?

ronaldbarendse commented:
I've redone my previous PR to be able to edit the language names in PR https://github.com/umbraco/Umbraco-CMS/pull/12243, but that does require some breaking changes (besides the fact it's a behavioral change). This also uses the `EnglishName` as the default language name at first, but then allows you to customize it, which I think is best of both worlds 😄 

Although the database schema isn't part of our official API surface, removing a column is kind of a breaking change, as any SQL query that fetches a language that includes all column names will break. Changing `DisplayName` to `EnglishName` is still a good fix we should ship in v9 though!

erikjanwestendorp commented:
@ronaldbarendse I just updated the PR. I deleted the migration (will create a new PR targeting the v10 branche). It now contains only the change from `DisplayName` to `EnglishName`.

ronaldbarendse commented:
> @ronaldbarendse I just updated the PR. I deleted the migration (will create a new PR targeting the v10 branche). It now contains only the change from `DisplayName` to `EnglishName`.

Looks good, just added 1 comment... And no need to retarget this for V10, as we will merge everything from V9.

Regarding https://github.com/umbraco/Umbraco-CMS/pull/12243, if users can edit the language name, they can use `#` prefixes to localize the names using dictionary items (similar to property names, descriptions, etc.). I haven't tested if that works, but should be trivial to implement and fix the issue in a more flexible way.

erikjanwestendorp commented on src/Umbraco.Infrastructure/Persistence/Factories/LanguageFactory.cs:
```diff
@@ -11,7 +11,7 @@ public static ILanguage BuildEntity(GlobalSettings globalSettings, LanguageDto d
             var lang = new Language(globalSettings, dto.IsoCode)
             {
-                CultureName = dto.CultureName,
+                CultureName = new CultureInfo(dto.IsoCode).EnglishName,
```
@ronaldbarendse Thanks again for your review. Just updated the PR.

erikjanwestendorp commented:
@ronaldbarendse @nul800sebastiaan @mikecp Thanks for your reviews. I'll close this one for now and create a new one targeting the v10 branche.

ronaldbarendse commented:
@erikjanwestendorp PR https://github.com/umbraco/Umbraco-CMS/pull/12243 already fixed this for v10 and I'm not sure whether changing `DisplayName` to `EnglishName` is something we still want to backport to v9 (maybe @nul800sebastiaan can give more feedback on that)?

Thanks for all the effort you've put into this PR though! :smile: If anything, this nudged me into redoing a v8 PR allowing you to edit the language names in v10 :+1:

nul800sebastiaan commented:
No at this point we're not backporting features to v9 any more. 

So I think this is all done now?

![image](https://user-images.githubusercontent.com/304656/178443042-a7a2c541-5eb8-4bc9-93d9-0ddaf7390ff2.png)
